src/main/java/brave/opentracing/BraveTracer.java

```diff
@@ -50,27 +52,65 @@
-
-  static final List<String> PROPAGATION_KEYS = Propagation.B3_STRING.keys();
-  static final TraceContext.Injector<TextMap> INJECTOR =
-      Propagation.B3_STRING.injector(TextMap::put);
-  static final TraceContext.Extractor<TextMapView> EXTRACTOR =
-      Propagation.B3_STRING.extractor(TextMapView::get);
-  static final Set<String> FIELDS_LOWER_CASE = lowercaseSet(PROPAGATION_KEYS);
-
-  private final brave.Tracer brave4;
-
   /**
    * Returns an implementation of {@linkplain io.opentracing.Tracer} which delegates
    * the the provided Brave Tracer.
    */
   public static BraveTracer wrap(brave.Tracer brave4) {
-    if (brave4 == null) throw new NullPointerException("brave tracer == null");
-    return new BraveTracer(brave4);
+    return newBuilder(brave4).build();
+  }
+
+  public static Builder newBuilder(brave.Tracer brave4) {
+    return new Builder(brave4);
+  }
+
+  public static final class Builder {
+    brave.Tracer brave4;
+    Map<Format<TextMap>, Propagation<String>> formatToPropagation = new LinkedHashMap<>();
+
+    Builder(brave.Tracer brave4) {
+      if (brave4 == null) throw new NullPointerException("brave tracer == null");
+      this.brave4 = brave4;
+      formatToPropagation.put(Format.Builtin.HTTP_HEADERS, Propagation.B3_STRING);
+      formatToPropagation.put(Format.Builtin.TEXT_MAP, Propagation.B3_STRING);
+    }
+
+    /**
+     * By default, {@link Format.Builtin#HTTP_HEADERS} and {@link Format.Builtin#TEXT_MAP} use
+     * {@link Propagation#B3_STRING B3 Propagation}. You can override or add different formats using
+     * this method.
+     *
+     * <p>For example, instead of using implicit format keys in your code, you might want to
+     * explicitly declare you are using B3. To do so, you'd do setup the tracer like this:
+     * <pre>{@code
+     * builder.propagation(MyFormats.B3, Propagation.B3_STRING);
+     *
+     * // later, you can ensure B3 is used like this:
+     * tracer.extract(MyFormats.B3, textMap);
+     * }</pre>
+     */
+    public Builder propagation(Format<TextMap> format, Propagation<String> propagation) {
+      if (format == null) throw new NullPointerException("format == null");
+      if (propagation == null) throw new NullPointerException("propagation == null");
+      formatToPropagation.put(format, propagation);
+      return this;
+    }
+
+    public BraveTracer build() {
+      return new BraveTracer(this);
+    }
   }
 
-  private BraveTracer(brave.Tracer brave4) {
-    this.brave4 = brave4;
+  final brave.Tracer brave4;
+  final Map<Format<TextMap>, Injector<TextMap>> formatToInjector = new LinkedHashMap<>();
+  final Map<Format<TextMap>, Extractor<TextMap>> formatToExtractor = new LinkedHashMap<>();
+
+  BraveTracer(Builder b) {
+    brave4 = b.brave4;
+    for (Map.Entry<Format<TextMap>, Propagation<String>> entry : b.formatToPropagation.entrySet()) {
+      formatToInjector.put(entry.getKey(), entry.getValue().injector(TextMap::put));
```

Comment:
My comment above would seem to complicate this, maybe its better to have builder methods for `Injector` and `Extractor` separately?